alephium/lib/deployment.ts

```diff
@@ -0,0 +1,437 @@
+import {
+  BuildDeployContractTx,
+  BuildExecuteScriptTx,
+  NodeProvider,
+  Contract,
+  Script,
+  Account,
+  Project,
+  Token,
+  CompilerOptions,
+  web3
+} from '@alephium/web3'
+import { PrivateKeyWallet } from '@alephium/web3-wallet'
+import path from 'path'
+import { waitTxConfirmed } from './utils'
+import fs, { promises as fsPromises } from 'fs'
+import * as cryptojs from 'crypto-js'
+
+export interface Network<Settings = unknown> {
+  networkId: number
+  nodeUrl: string
+  mnemonic: string
+  deploymentFile: string
+  confirmations: number
+  settings: Settings
+}
+
+export type NetworkType = 'mainnet' | 'testnet' | 'devnet'
+
+export interface Configuration<Settings = unknown> {
+  sourcePath?: string
+  artifactPath?: string
+
+  deployScriptsPath: string
+  compilerOptions: CompilerOptions
```

Comment:
Should we make both optional too? The default value for them work in most cases.